### DISCUSSION TITLE
Content Create API "content/content/v1/create" is failing

### DISCUSSION THREAD

**Discussion by komalm** [Sunbird-Knowlg/Community#24] [Q&A] [answered]
Hi @keshavprasadms
Hope you are doing well!..

**Site Environments Details** - SunbirdEd portal of version 4.4.1

**Issue** - Content Create API "content/content/v1/create" is failing

**Debugging Result -**

```
Request URL: {{host}}/content/content/v1/create
Request Method: POST
Status Code: 400 Bad Request
```

**Response -**

```
{
    "id": "api.content.create",
    "ver": "1.0",
    "ts": "2022-06-19T16:54:44.869Z",
    "params": {
        "resmsgid": "85447b50-eff0-11ec-9195-03ede13ce066",
        "msgid": null,
        "status": "failed",
        "err": "ERR_EMPTY_EDGE_PROPERTY_LIST",
        "errmsg": "The list to validate input is empty."
    },
    "responseCode": "CLIENT_ERROR",
    "result": {
        "messages": null
    }
}
```

**Note -** The License are created and Default License is assigned to the channel. When we check the read channel API, we can see the Default License added to that channel.

**Read Channel API response -** 

```
createdOn: "2020-05-06T13:02:17.752+0000"
defaultCourseFramework: "TPD"
defaultFramework: "{{defaultFramework}}"
defaultLicense: "CC BY-NC-SA 4.0"
description: "This Org description"
```

Can you please help us with the same?

cc @coolbung, @Renuka-S, @vinod303

**Comment by saiakhil46** [comment]
@komalm After updating redis with `redis-cli SADD edge_license "CC BY 4.0"` it started working

`CC BY 4.0` is default license for content

**Reply by coolbung** [reply]
@amit-tarento is this a bug ? Shouldn't the cache be pass-through i.e. the cache should automatically populate if it's not populated, rather than throwing an error ?

Also, is there a job to populate these caches ?

**Comment by komalm** [comment, marked as answer]
Thank's @saiakhil46, issue is resolved. 

We have followed the below steps to solve the issue in this case - 

1. Check to see whether the licenses are available in the system (http://docs.sunbird.org/latest/apis/license/)
2. Check Default License is available for the channel (http://docs.sunbird.org/3.5.0/developer-docs/server-installation/post_installation_configuration/)
3. If **step-1** and **step-2** are validated and the license is present, then check the Redis Cache whether the license details are present in the REDIS.
If not, as mentioned by @saiakhil46  in the [previous comment](https://github.com/orgs/Sunbird-Knowlg/discussions/24#discussioncomment-2984409), update the license with the relevant license. 
`
eg. redis-cli SADD edge_license "CC BY 4.0" 
`
(as done by @saiakhil46  in this case)
